PORTAL/jobs/__main__.py

```diff
@@ -213,19 +251,24 @@ def cmd_cancel(jobs, reqid=None, *, _status=None):
         else:
             cmd_queue_remove(jobs, reqid)
             job = jobs.get(reqid)
-            job.cancel(ifstatus=_status)
+            if job:
+                job.cancel(ifstatus=_status)
+            else:
+                raise RuntimeError(f"Couldn't get job for {reqid}")
 
     logger.info('')
     logger.info('Results:')
     # XXX Show something better?
-    for line in job.render(fmt='resfile'):
-        logger.info(line)
+
+    if job:
```

Comment:
Yeah, that makes sense.  There is a code path where it can be undefined, though -- if [this line raises an exception](https://github.com/faster-cpython/tools/blob/main/PORTAL/jobs/__main__.py#L210).  I guess there should be a return in that `except` block?  I'll do that for now...